packages/components/src/DataList/DataList.test.tsx

```diff
@@ -5,12 +5,101 @@ import {
   EMPTY_FILTER_RESULTS_ACTION_LABEL,
   EMPTY_FILTER_RESULTS_MESSAGE,
 } from "./DataList.const";
-import { DataListProps } from "./DataList.types";
+import { DataListItemType, DataListProps } from "./DataList.types";
+import { Grid } from "../Grid";
+import { GLIMMER_TEST_ID } from "../Glimmer";
 
 describe("DataList", () => {
+  it("should render the total count", () => {
+    const mappedData = [{ label: "Name" }];
+    render(
+      <DataList
+        loading={false}
+        totalCount={10}
+        data={[{ label: "Luke Skywalker" }]}
+        headers={{
+          label: "Name",
+        }}
+      >
+        <DataList.Layout>
+          {(item: DataListItemType<typeof mappedData>) => (
+            <Grid alignItems="center">
+              <Grid.Cell size={{ xs: 6 }}>{item.label}</Grid.Cell>
+            </Grid>
+          )}
+        </DataList.Layout>
+      </DataList>,
+    );
+    expect(screen.getByText("(10 results)")).toBeInTheDocument();
+  });
+
+  it("should not render the total count if the totalCount is null", () => {
+    const mappedData = [{ label: "Name" }];
+    render(
+      <DataList
+        loading={false}
+        totalCount={null}
+        data={[{ label: "Luke Skywalker" }]}
+        headers={{
+          label: "Name",
+        }}
+        title="All Clients"
+      >
+        <DataList.Layout>
+          {(item: DataListItemType<typeof mappedData>) => (
+            <Grid alignItems="center">
+              <Grid.Cell size={{ xs: 6 }}>{item.label}</Grid.Cell>
+            </Grid>
+          )}
+        </DataList.Layout>
+      </DataList>,
+    );
+    expect(screen.getByText("All Clients")).toBeInTheDocument();
+    expect(screen.queryByText("(10 results)")).toBeNull();
+  });
+
+  it("should render the Glimmer when loading", () => {
```

Comment:
Keep in mind that the moment you do the loading ticket, this will break. Is this testing the loading in general? Or the loading of the totalCount?

Reply:
I added a test ID and queried within it. What do you think?